tests/unit/test_gate_requirements.py

```
@@ -659,3 +659,113 @@ def get_check(self, name, cfg):
+        _print_required_deps(tmp_path)
+        first = capsys.readouterr().out
+        _print_required_deps(tmp_path)
+        second = capsys.readouterr().out
+        assert first == second  # byte-stable for a fixed config
+
+    def test_aggregate_raises_on_conflicting_declarations(self, monkeypatch):
+        from unittest.mock import MagicMock
+
+        from slopmop.checks import tool_inventory
+
+        def gate(*reqs):
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `gate`

(ANN202)